File: HW10/LevelOrderMain.py
from collections import deque
from typing import List

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class LevelOrderMain:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        finalAns = []
        if not root:
            return finalAns
        queue = deque([root])
        while queue:
            levels = len(queue)
            subLevels = []
            for _ in range(levels):
                node = queue.popleft()
                subLevels.append(node.val)
                if node.left:
                    queue.append(node.left)
                if node.right:
                    queue.append(node.right)
            finalAns.append(subLevels)
        return finalAns

if __name__ == "__main__":
    tree = LevelOrderMain()
    tree.root = TreeNode(18)
    tree.root.left = TreeNode(8)
    tree.root.left.left = TreeNode(7)
    tree.root.left.right = TreeNode(9)
    tree.root.right = TreeNode(15)
    tree.root.right.left = TreeNode(14)
    tree.root.right.right = TreeNode(16)
    values = tree.levelOrder(tree.root)
    print(values)

File: HW10/LevelOrderMain.java
import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;
import java.util.Queue;

public class LevelOrderMain {
	TreeNode root;

	public List<List<Integer>> levelOrder(TreeNode root) {
		Queue<TreeNode> queue = new LinkedList<>();
		List<List<Integer>> finalAns = new ArrayList<List<Integer>>();
		if (root == null) {
			return finalAns;
		}
		queue.add(root);
		while (!queue.isEmpty()) {
			int levels = queue.size();
			List<Integer> subLevels = new ArrayList<>();
			for (int i = 0; i < levels; i++) {
				if (queue.peek().left != null) {
					queue.add(queue.peek().left);
				}
				if (queue.peek().right != null) {
					queue.add(queue.peek().right);
				}
				subLevels.add(queue.remove().val);
			}
			finalAns.add(subLevels);
		}
		return finalAns;
	}

	public static void main(String[] args) {
		LevelOrderMain tree = new LevelOrderMain();
		tree.root = new TreeNode(18);
		tree.root.left = new TreeNode(8);
		tree.root.left.left = new TreeNode(7);
		tree.root.left.right = new TreeNode(9);
		tree.root.right = new TreeNode(15);
		tree.root.right.left = new TreeNode(14);
		tree.root.right.right = new TreeNode(16);
		List<List<Integer>> values = tree.levelOrder(tree.root);

		System.out.println(values);

	}

}
